### PR TITLE
Fedora40

### PR DESCRIPTION
**Pull request by cevich**
Depends on https://github.com/containers/automation_images/pull/348

### CODE REVIEW

**Comment by cevich** [comment]
@edsantiago I could use your informed opinion here.  The "Bump up to CentOS Stream 9 tooling images" commit seems pertinent to this PR (IIUC CS8 is going EoL soonish?).  However it's breaking the image-build process in a difficult-to-debug way.  Do you think the need for F40 CI VM images is urgent enough I should split that commit into a dedicated PR?

**Comment by edsantiago** [comment]
I'm always a fan of breaking down big changes into little ones, especially when they run into problems. That said, I don't see an urgent need right now for f40 VMs. It'll probably be fairly close to what we're using for rawhide.

**Comment by cevich** [comment]
> especially when they run into problems.

Thanks for your opinion.  I think I'll give it a bit more debugging effort, then split if I don't make significant forward progress.

> It'll probably be fairly close to what we're using for rawhide.

Oh, true that.

Thanks Ed

**Comment by edsantiago** [comment]
- 20240429t164200z-f40f39d13
- 20240411t124913z-f39f38d13 &#x21D1; (built in #340)

|  |*debian*|*prior-fedora*|*fedora*|*fedora-aws*|*rawhide*|
|----|----|----|----|----|----|
|*base*         | 13.1            | 39-1.5          | Generic         | ?               | 41-0            |
|               | 13&nbsp;&#x21D1; | 38-1.6&nbsp;&#x21D1; | 39-1.5&nbsp;&#x21D1; |                |                |
|*kernel*       | 6.7.12-1        | 6.8.7-200       | 6.8.8-300       | 6.8.7-300       | 6.9.0-0.rc5.20240426gitc942a0cd3603.48 |
|               | 6.7.9-2&nbsp;&#x21D1; | 6.8.4-100&nbsp;&#x21D1; | 6.8.5-200&nbsp;&#x21D1; | 6.8.4-200&nbsp;&#x21D1; | 6.9.0-0.rc3.30&nbsp;&#x21D1; |
|*grub2-common* | 2.12-2          | 2.06-120        | 2.06-121        | 2.06-121        | 2.06-121        |
|               |                | 2.06-116&nbsp;&#x21D1; | 2.06-118&nbsp;&#x21D1; | 2.06-118&nbsp;&#x21D1; | 2.06-119&nbsp;&#x21D1; |
|*aardvark-dns* | 1.4.0-5         | 1.10.0-1        | 1.10.0-1        | 1.10.0-1        | 1.10.0-1        |
|               |                |                |                |                |                |
|*netavark*     | 1.4.0-4         | 1.10.3-1        | 1.10.3-3        | 1.10.3-3        | 1.10.3-3        |
|               |                |                | 1.10.3-1&nbsp;&#x21D1; | 1.10.3-1&nbsp;&#x21D1; |                |
|*buildah*      | 1.33.7+ds1-1    | 1.35.3-1        | 1.35.3-1        | 1.35.3-1        | 1.35.3-1        |
|               |                |                |                |                |                |
|*conmon*       | 2.1.10+ds1-1+b1 | 2.1.10-1        | 2.1.10-1        | 2.1.10-1        | 2.1.10-1        |
|               |                |                |                |                |                |
|*container-selinux*| ?               | 2.230.0-1       | 2.231.0-1       | 2.230.0-1       | 2.231.0-1       |
|               |                |                | 2.230.0-1&nbsp;&#x21D1; |                | 2.230.0-1&nbsp;&#x21D1; |
|*containers-common*| ?               | 1-99            | 0.58.0-2        | 0.58.0-2        | 0.58.0-18       |
|               |                | 1-89&nbsp;&#x21D1; | 1-99&nbsp;&#x21D1; | 1-99&nbsp;&#x21D1; | 0.58.0-8&nbsp;&#x21D1; |
|*criu*         | 3.17.1-3        | 3.19-2          | 3.19-4          | 3.19-4          | 3.19-4          |
|               |                | 3.18-1&nbsp;&#x21D1; | 3.19-2&nbsp;&#x21D1; | 3.19-2&nbsp;&#x21D1; |                |
|*crun*         | 1.14.4-1        | 1.14.4-1        | 1.14.4-1        | 1.14.4-1        | 1.14.4-5        |
|               |                |                |                |                |                |
|*golang*       | 2:1.22~3        | 1.21.9-1        | 1.22.2-1        | 1.22.2-1        | 1.22.2-3        |
|               |                | 1.21.8-1&nbsp;&#x21D1; | 1.21.9-1&nbsp;&#x21D1; | 1.21.9-1&nbsp;&#x21D1; |                |
|*gvisor-tap-vsock*| ?               | 0.7.3-1         | 0.7.3-2         | 0.7.3-2         | 0.7.3-2         |
|               |                |                | 0.7.3-1&nbsp;&#x21D1; | 0.7.3-1&nbsp;&#x21D1; |                |
|*nmap-ncat*    | 7.94+git20230807.3be01efb1+dfsg-3+b1 | 7.94-1          | 7.95-1          | 7.95-1          | 7.95-1          |
|               |                | 7.93-2&nbsp;&#x21D1; | 7.94-1&nbsp;&#x21D1; | 7.94-1&nbsp;&#x21D1; | 7.94-1&nbsp;&#x21D1; |
|*passt*        | 2024-04-26      | 2024-04-05      | 2024-04-26      | 2024-04-05      | 2024-04-26      |
|               | 2024-03-26&nbsp;&#x21D1; |                | 2024-04-05&nbsp;&#x21D1; |                | 2024-04-05&nbsp;&#x21D1; |
|*podman*       | 4.9.4+ds1-1     | 4.9.4-1         | 5.0.2-1         | 5.0.2-1         | 5.0.2-1         |
|               |                |                | 4.9.4-1&nbsp;&#x21D1; | 4.9.4-1&nbsp;&#x21D1; | 5.0.1-1&nbsp;&#x21D1; |
|*runc*         | 1.1.12+ds1-2    | 1.1.12-1        | 1.1.12-3        | 1.1.12-3        | 1.1.12-3        |
|               |                |                | 1.1.12-1&nbsp;&#x21D1; | 1.1.12-1&nbsp;&#x21D1; |                |
|*skopeo*       | 1.13.3+ds1-2+b1 | 1.15.0-1        | 1.15.0-1        | 1.15.0-1        | 1.15.0-1        |
|               |                |                |                |                |                |
|*slirp4netns*  | 1.2.1-1+b1      | 1.2.2-1         | 1.2.2-2         | 1.2.2-2         | 1.2.2-2         |
|               |                |                | 1.2.2-1&nbsp;&#x21D1; | 1.2.2-1&nbsp;&#x21D1; |                |
|*systemd*      | 255.5-1         | 254.10-1        | 255.4-1         | 255.4-1         | 255.5-1         |
|               | 255.4-1+b1&nbsp;&#x21D1; | 253.17-1&nbsp;&#x21D1; | 254.10-1&nbsp;&#x21D1; | 254.10-1&nbsp;&#x21D1; | 255.4-1&nbsp;&#x21D1; |
|*tar*          | 1.34+dfsg-1.2+deb12u1 | 1.35-2          | 1.35-3          | 1.35-3          | 1.35-3          |
|               |                | 1.34-8&nbsp;&#x21D1; | 1.35-2&nbsp;&#x21D1; | 1.35-2&nbsp;&#x21D1; |                |

**Comment by cevich** [comment]
Holy crap it worked!

**Comment by cevich** [comment]
@edsantiago I'd appreciate your input on this, I'm thinking maybe it should be the next tagged release?  I can't make sense of the version table, are there any action-items I should take based on it?

I see there are still "ticking" timebombs for pasta stabilization in Fedora.  Then a tar related issue in debian it looks like the version went backwards?  Maybe they fixed it?

**Comment by cevich** [comment]
> Tagged release: oh please no, not without at least submitting a test PR on podman.

Oh sorry, I was assuming testing in podman.

> white lines show current package version, gray lines show prior version.

Ooooohhh, I didn't understand that part.

> debian tar: According to my table there is no change in debian tar?

It's my fault, I didn't realize the table showed current and prior versions.  I thought it was saying `1.34+dfsg-1.2+deb12u1` was the current version.

Thanks for the feedback Ed, I'll start running this against podman...

**Comment by Luap99** [comment]
```
setting up Pasta: pasta failed with exit code 1:
           Couldn't open network namespace /var/tmp/podman-e2e-3444609741/subtest-884223713/runroot/networks/rootless-netns/rootless-netns: Permission denied
```
pasta is not working on debian, I bet apparmor is blocking access
https://github.com/containers/buildah/issues/5440

Although the version looks new enough and given the old version worked this seems weird??

**Comment by edsantiago** [comment]
FWIW apparmor itself appears to be unchanged: `3.0.13-2` in both the old VMs and the new ones. Pasta bumped from 03-26 to 04-26 in debian as shown in the table above.

**Comment by cevich** [comment]
> FWIW apparmor itself appears to be unchanged

I know nothing about it, but I wonder if there's a separate "rules" package similar to how we have "containers-selinux" in Fedora?  Or maybe the rules ship with the parent package? 
In any case perhaps that's the thing in need of updating?

**Comment by edsantiago** [comment]
My script downloads all the logs (base & cache). I grep'ed for `passt` in all debian logs for 202404, and the only match is passt itself. (i.e., no "passt-apparmor" or anything like that). Grepping for `apparmor` produces more hits, but those are "libapparmor" and "libapparmor-dev", and are consistently 3.0.13-2, and are unlikely to have special rules for pasta.

**Comment by Luap99** [comment]
I know what the issue is so don't waste to much time on it, I don't know why it worked before though.
We have to tweak the apparmor profile to allow our tmpdir test paths, I look into it on Thursday

**Comment by cevich** [comment]
> I know what the issue is so don't waste to much time on it

Oh great, I'm glad to hear that Paul.  Previously I believe the Debian image was on a much older version according to Ed's table.  Thursday is fine, and thanks for the heads up.  There's no supreme rush for these images that I'm aware of.

**Comment by sbrivio-rh** [comment]
> I know what the issue is so don't waste to much time on it, I don't know why it worked before though.

Oops, that's because the package didn't install the AppArmor profile for pasta(1). I just [added that](https://salsa.debian.org/sbrivio/passt/-/commit/4a77ef55c34c579d4845aa2dfd003abf2195ea9b) to unstable, and boom. Sorry.

> We have to tweak the apparmor profile to allow our tmpdir test paths, I look into it on Thursday

Right, I should use the `user-tmp` abstraction (which enables access to `/var/tmp/**` among others) instead of just granting access to `/tmp/**` as I did. If that's the only problematic thing there, I can take care of it too.

**Comment by cevich** [comment]
Thanks @sbrivio-rh

> I just added that to unstable

Please forgive my ignorance but does that mean it's already landed in Debian SID or we need to wait for that?

For reference, I just tried installing passt in a SID container image and got `0.0~git20240426.d03c4e2-1`

**Comment by sbrivio-rh** [comment]
> Please forgive my ignorance but does that mean it's already landed in Debian SID or we need to wait for that?

That means I committed this change and then prepared a new package for unstable including it, so it's already there, exactly in:

> For reference, I just tried installing passt in a SID container image and got `0.0~git20240426.d03c4e2-1`

this version, `0.0~git20240426.d03c4e2-1`. See also the [changelog](https://tracker.debian.org/media/packages/p/passt/changelog-0.0git20240426.d03c4e2-1) -- you find all the link from the "versioned links" section of the [tracker](https://tracker.debian.org/pkg/passt), if you're curious.

This new version is what gives you the issue with the AppArmor profile (not a separate package) not allowing pasta to access the namespace link at `/var/tmp/podman-e2e-...`.

Before this version, by mistake, the package wouldn't install the AppArmor profile for pasta, so you wouldn't see any issue. I'll have to fix it and prepare a new version.

**Comment by cevich** [comment]
Thanks for the details @sbrivio-rh I'll trigger a new build of our Debian SID CI VM images.

**Comment by sbrivio-rh** [comment]
> Thanks for the details @sbrivio-rh I'll trigger a new build of our Debian SID CI VM images.

Wait, that won't work anyway! I'll have to fix the AppArmor profile first:

>>  We have to tweak the apparmor profile to allow our tmpdir test paths, I look into it on Thursday
>
> Right, I should use the user-tmp abstraction (which enables access to /var/tmp/** among others) instead of just granting access to /tmp/** as I did. If that's the only problematic thing there, I can take care of it too.

For clarity:

- old versions (pre-`0.0~git20240426.d03c4e2-1`): no AppArmor profile installed, no CI issue
- current version `0.0~git20240426.d03c4e2-1`: AppArmor profile installed, too restrictive, CI issue appeared
- next version (not done yet): AppArmor profile will be less restrictive and fix the CI issue

**Comment by cevich** [comment]
> Wait, that won't work anyway! I'll have to fix the AppArmor profile first:

Oh my bad, I already started it.  No worries, there's a bunch of other workarounds I'd like to test removing.  I'll wait for your "go ahead" WRT debian passt before starting any podman CI testing on these images.  Thanks again.

File: image_builder/google-cloud-sdk.repo [hunk 1]
```diff
@@ -1,16 +1,7 @@
-[google-compute-engine]
-name=Google Compute Engine
-baseurl=https://packages.cloud.google.com/yum/repos/google-compute-engine-el8-x86_64-stable
+[google-cloud-cli]
```
Comment (cevich) [inline image_builder/google-cloud-sdk.repo:1]: My next push will include a comment indicating where the contents of this file came from.

File: imgts/google-cloud-sdk.repo [hunk 2]
```diff
@@ -1,19 +1,7 @@
...
-gpgcheck=1
-repo_gpgcheck=1
-gpgkey=https://packages.cloud.google.com/yum/doc/yum-key.gpg
-       https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
-
-# From https://cloud.google.com/sdk/docs/install#rpm
-[google-cloud-sdk]
-name=Google Cloud SDK
-baseurl=https://packages.cloud.google.com/yum/repos/cloud-sdk-el8-x86_64
+[google-cloud-cli]
```
Comment (cevich) [inline imgts/google-cloud-sdk.repo:1]: My next push will include a comment indicating where the contents of this file came from.

**Comment by edsantiago** [comment]
- 20240501t194928z-f40f39d13
- 20240411t124913z-f39f38d13 &#x21D1; (built in #340)

|  |*debian*|*prior-fedora*|*fedora*|*fedora-aws*|*rawhide*|
|----|----|----|----|----|----|
|*base*         | 13.1            | 39-1.5          | Generic         | ?               | 41-0            |
|               | 13&nbsp;&#x21D1; | 38-1.6&nbsp;&#x21D1; | 39-1.5&nbsp;&#x21D1; |                |                |
|*kernel*       | 6.7.12-1        | 6.8.7-200       | 6.8.8-300       | 6.8.7-300       | 6.9.0-0.rc6.52  |
|               | 6.7.9-2&nbsp;&#x21D1; | 6.8.4-100&nbsp;&#x21D1; | 6.8.5-200&nbsp;&#x21D1; | 6.8.4-200&nbsp;&#x21D1; | 6.9.0-0.rc3.30&nbsp;&#x21D1; |
|*grub2-common* | 2.12-2          | 2.06-120        | 2.06-121        | 2.06-121        | 2.06-121        |
|               |                | 2.06-116&nbsp;&#x21D1; | 2.06-118&nbsp;&#x21D1; | 2.06-118&nbsp;&#x21D1; | 2.06-119&nbsp;&#x21D1; |
|*aardvark-dns* | 1.4.0-5         | 1.10.0-1        | 1.10.0-1        | 1.10.0-1        | 1.10.0-1        |
|               |                |                |                |                |                |
|*netavark*     | 1.4.0-4         | 1.10.3-1        | 1.10.3-3        | 1.10.3-3        | 1.10.3-3        |
|               |                |                | 1.10.3-1&nbsp;&#x21D1; | 1.10.3-1&nbsp;&#x21D1; |                |
|*buildah*      | 1.33.7+ds1-1    | 1.35.3-1        | 1.35.3-1        | 1.35.3-1        | 1.35.3-1        |
|               |                |                |                |                |                |
|*conmon*       | 2.1.10+ds1-1+b1 | 2.1.10-1        | 2.1.10-1        | 2.1.10-1        | 2.1.10-1        |
|               |                |                |                |                |                |
|*container-selinux*| ?               | 2.230.0-1       | 2.231.0-1       | 2.230.0-1       | 2.231.0-1       |
|               |                |                | 2.230.0-1&nbsp;&#x21D1; |                | 2.230.0-1&nbsp;&#x21D1; |
|*containers-common*| ?               | 1-99            | 0.58.0-2        | 0.58.0-2        | 0.58.0-18       |
|               |                | 1-89&nbsp;&#x21D1; | 1-99&nbsp;&#x21D1; | 1-99&nbsp;&#x21D1; | 0.58.0-8&nbsp;&#x21D1; |
|*criu*         | 3.17.1-3        | 3.19-2          | 3.19-4          | 3.19-4          | 3.19-4          |
|               |                | 3.18-1&nbsp;&#x21D1; | 3.19-2&nbsp;&#x21D1; | 3.19-2&nbsp;&#x21D1; |                |
|*crun*         | 1.14.4-1        | 1.14.4-1        | 1.14.4-1        | 1.14.4-1        | 1.14.4-5        |
|               |                |                |                |                |                |
|*golang*       | 2:1.22~3        | 1.21.9-1        | 1.22.2-1        | 1.22.2-1        | 1.22.2-3        |
|               |                | 1.21.8-1&nbsp;&#x21D1; | 1.21.9-1&nbsp;&#x21D1; | 1.21.9-1&nbsp;&#x21D1; |                |
|*gvisor-tap-vsock*| ?               | 0.7.3-1         | 0.7.3-2         | 0.7.3-2         | 0.7.3-2         |
|               |                |                | 0.7.3-1&nbsp;&#x21D1; | 0.7.3-1&nbsp;&#x21D1; |                |
|*nmap-ncat*    | 7.94+git20230807.3be01efb1+dfsg-3+b1 | 7.94-1          | 7.95-1          | 7.95-1          | 7.95-1          |
|               |                | 7.93-2&nbsp;&#x21D1; | 7.94-1&nbsp;&#x21D1; | 7.94-1&nbsp;&#x21D1; | 7.94-1&nbsp;&#x21D1; |
|*passt*        | 2024-04-26      | 2024-04-26      | 2024-04-26      | 2024-04-26      | 2024-04-26      |
|               | 2024-03-26&nbsp;&#x21D1; | 2024-04-05&nbsp;&#x21D1; | 2024-04-05&nbsp;&#x21D1; | 2024-04-05&nbsp;&#x21D1; | 2024-04-05&nbsp;&#x21D1; |
|*podman*       | 4.9.4+ds1-1     | 4.9.4-1         | 5.0.2-1         | 5.0.2-1         | 5.0.2-1         |
|               |                |                | 4.9.4-1&nbsp;&#x21D1; | 4.9.4-1&nbsp;&#x21D1; | 5.0.1-1&nbsp;&#x21D1; |
|*runc*         | 1.1.12+ds1-2    | 1.1.12-1        | 1.1.12-3        | 1.1.12-3        | 1.1.12-3        |
|               |                |                | 1.1.12-1&nbsp;&#x21D1; | 1.1.12-1&nbsp;&#x21D1; |                |
|*skopeo*       | 1.13.3+ds1-2+b1 | 1.15.0-1        | 1.15.0-1        | 1.15.0-1        | 1.15.0-1        |
|               |                |                |                |                |                |
|*slirp4netns*  | 1.2.1-1+b1      | 1.2.2-1         | 1.2.2-2         | 1.2.2-2         | 1.2.2-2         |
|               |                |                | 1.2.2-1&nbsp;&#x21D1; | 1.2.2-1&nbsp;&#x21D1; |                |
|*systemd*      | 255.5-1         | 254.10-1        | 255.4-1         | 255.4-1         | 255.5-1         |
|               | 255.4-1+b1&nbsp;&#x21D1; | 253.17-1&nbsp;&#x21D1; | 254.10-1&nbsp;&#x21D1; | 254.10-1&nbsp;&#x21D1; | 255.4-1&nbsp;&#x21D1; |
|*tar*          | 1.35+dfsg-3     | 1.35-2          | 1.35-3          | 1.35-3          | 1.35-3          |
|               | 1.34+dfsg-1.2+deb12u1&nbsp;&#x21D1; | 1.34-8&nbsp;&#x21D1; | 1.35-2&nbsp;&#x21D1; | 1.35-2&nbsp;&#x21D1; |                |

**Comment by Luap99** [comment]
> > I know what the issue is so don't waste to much time on it, I don't know why it worked before though.
> 
> Oops, that's because the package didn't install the AppArmor profile for pasta(1). I just [added that](https://salsa.debian.org/sbrivio/passt/-/commit/4a77ef55c34c579d4845aa2dfd003abf2195ea9b) to unstable, and boom. Sorry.

Thank @sbrivio-rh, that was the missing I piece I did not grasp.

I must say the timing here was quite unfortunate, until not long ago we always used /tmp/.. in CI testing so it would not have caused issues AFAICT. Then we changed to /var/tmp as out tests exceeded tmpfs limits but now I already try to get them back onto /tmp tmpfs to speed them up.

@sbrivio-rh To unblock us without waiting for a new pasta package I guess we can extend the pasta profile ourselves here?

**Comment by sbrivio-rh** [comment]
> @sbrivio-rh To unblock us without waiting for a new pasta package I guess we can extend the pasta profile ourselves here?

Ah, yes, that would be nice, it would relieve me from the pressure of preparing yet a new version for Debian (I'm still a sponsored maintainer, there -- I don't have upload rights _yet_).

It should simply be this: instead of `owner /tmp/**` in `/etc/apparmor.d/usr.bin.pasta`, use `include <abstractions/user-tmp>`.

It's also indicated by a comment there, but I never switched to `user-tmp` because I didn't see the point of permitting access to `/var/tmp`, and here it comes.

**Comment by Luap99** [comment]
@cevich Add something like this and maybe a timebomb to not forget it to remove it in the future when there is a new upgrade with a updated profile.
```
cat > /etc/apparmor.d/usr.bin.pasta <<EOF
abi <abi/3.0>,

include <tunables/global>

profile pasta /usr/bin/pasta{,.avx2} flags=(attach_disconnected) {
  include <abstractions/pasta>
  include <abstractions/user-tmp>

  owner @{HOME}/**			w,	# pcap(), write_pidfile()
}
EOF
```

**Comment by cevich** [comment]
> I must say the timing here was quite unfortunate

@Luap99 another way of looking at it is @edsantiago work moving us to /var/tmp helped us ***FIND*** a bug :grin: 

> Add something like this and maybe a timebomb

Yep, on it.  I was going to try and figure it out myself, but since you provided some copy-paste, I will happily oblige :grinning:  

Any reason to believe that temporary workaround will clash with the one from the updated package when it arrives?  (that would be better than a timebomb() but I'll add one also)

**Comment by edsantiago** [comment]
This has grown too much. Could you please split it into smaller PRs?

**Comment by cevich** [comment]
> This has grown too much. Could you please split it into smaller PRs?

I'll think about it, esp. moreso if it runs into any complex problems during testing.

**Comment by cevich** [comment]
Well of course [the Windows build broke](https://cirrus-ci.com/task/4805889931083776).  S'pose that's as good an excuse as any to split this PR up.

**Comment by edsantiago** [comment]
Comparing to previous build _in this PR_, for clarity

- 20240502t200454z-f40f39d13
- 20240501t194928z-f40f39d13 &#x21D1; (built in #349)

|  |*debian*|*prior-fedora*|*fedora*|*fedora-aws*|*rawhide*|
|----|----|----|----|----|----|
|*aardvark-dns* | 1.4.0-5.1       | 1.10.0-1        | 1.10.0-1        | 1.10.0-1        | 1.10.0-1        |
|               | 1.4.0-5&nbsp;&#x21D1; |                |                |                |                |
|*nmap-ncat*    | 7.94+git20230807.3be01efb1+dfsg-3+b1 | 7.95-1          | 7.95-1          | 7.95-1          | 7.95-1          |
|               |                | 7.94-1&nbsp;&#x21D1; |                |                |                |

**Comment by cevich** [comment]
Oh that worked!

Re: Simplifying this PR

Okay I see what's going on.  I remember there's a delay of some hours before a GCP image push registers as the "latest" in the family.  So the CentOS 9 update isn't actually required in this PR. But when I moved it out, at least for a time, builds were failing because they were (unintentionally) running on CentOS 9 w/o the packer/qemu changes.  It may have even been picking up the IBI image from #352 since I started builds there as well.

Moral of the story: CI VM builds can't work reliably/predictably when they're happening on multiple PRs at the same time.  This could be fixed (in theory) by not referencing the IBI by family.  However there was a good reason I set it that way at one point, and I don't feel like digging today.  We also don't change the IBI internals frequently, so it's probably not worth sorting out.

I'm keen to just leave the CentOS 9 changes in this PR then, unless there's a strong objection.  It may take a full day of playing games with PRs to get the CentOS 8 IBI back for use here, before it's just replaced again by another PR after this merges :confused:

**Comment by cevich** [comment]
@Luap99 On the `20240502t200454z-f40f39d13` debian image, I find:

```
# ls -la /etc/apparmor.d
...cut...
-rw-r--r--  1 root root   676 Apr 26 05:46 usr.bin.passt
-rw-r--r--  1 root root   232 May  2 20:57 usr.bin.pasta
...cut...
```

is this expected?

The `passt` file contains:

```
abi <abi/3.0>,

include <tunables/global>

profile passt /usr/bin/passt{,.avx2} {
  include <abstractions/passt>

  # Alternatively: include <abstractions/user-tmp>
  owner /tmp/**                         w,      # tap_sock_unix_init(), pcap(),
                                                # write_pidfile(),
                                                # logfile_init()

  owner @{HOME}/**                      w,      # pcap(), write_pidfile()
}
```

**Comment by sbrivio-rh** [comment]
> > I must say the timing here was quite unfortunate
> 
> @Luap99 another way of looking at it is @edsantiago work moving us to /var/tmp helped us _**FIND**_ a bug 😁

I forgot to answer here: I wouldn't be so sure that's a bug :) ...we didn't get any report of any other user using /var/tmp/ for namespace links. So, sure, let's "fix" this because it's annoying for this CI, but whether /var/tmp/ is a "reasonable" path is not so obvious.

> @Luap99 On the 20240502t200454z-f40f39d13 debian image, I find:
> ```
> # ls -la /etc/apparmor.d
> ...cut...
> -rw-r--r--  1 root root   676 Apr 26 05:46 usr.bin.passt
> -rw-r--r--  1 root root   232 May  2 20:57 usr.bin.pasta
> ...cut...
> ```
> is this expected?

The content of what @Luap99 posted in https://github.com/containers/automation_images/pull/349#issuecomment-2090535529 is 228 bytes, not 232. It looks like there are 4 extra bytes in `usr.bin.pasta`?

> The passt file contains:

That's for VMs (passt(1)). What you want to edit is `usr.bin.pasta`, which is for containers (pasta(1)).

> Any reason to believe that temporary workaround will clash with the one from the updated package when it arrives? (that would be better than a timebomb() but I'll add one also)

It won't clash, but I think it would be good to drop the workaround at some point so that we go back to test what the package actually ships.

**Comment by cevich** [comment]
Thanks @sbrivio-rh.

> but I think it would be good to drop the workaround

Yep, I put a thing in place to remind us to circle back here in a month or so.

> is 228 bytes, not 232

I think the contents are correct, (not sure why the byte discrepancy, no newline at the end or some encoding difference?):

```
# cat /etc/apparmor.d/usr.bin.pasta
abi <abi/3.0>,

include <tunables/global>

profile pasta /usr/bin/pasta{,.avx2} flags=(attach_disconnected) {
  include <abstractions/pasta>
  include <abstractions/user-tmp>

  owner @{HOME}/**      w,  # pcap(), write_pidfile()
}
```

(Copy-pasted right from the CI VM I built)

**Comment by edsantiago** [comment]
Debian tar is [1.35](https://github.com/containers/automation_images/pull/349#issuecomment-2089192369) which is known to be broken. Looking at your diffs it looks like the timebomb should work, i.e. you *should* be blocking 1.35. I will look into it during the work week. In a nutshell, though, Debian tar 1.35 is 100% guaranteed to fail.

**Comment by edsantiago** [comment]
It's really hard to debug this, but the bug is in [debian base](https://cirrus-ci.com/task/5812336529768448?logs=main#L917):
```
    debian: Preparing to unpack .../tar_1.35+dfsg-3_amd64.deb ...
    debian: Unpacking tar (1.35+dfsg-3) over (1.34+dfsg-1.2+deb12u1) ...
```
Best I can tell, this is happening in `install_automation_tooling` which I wasn't able to dive into deeply enough to figure out. But if this hypothesis is true, perhaps the solution is to move the debian-tar timebomb up a few lines, to earlier in `debian_base_setup.sh`.

Also, I belatedly realized that we really, really need a debugging printf in `timebomb()`, something that will add a log entry, but `timebomb()` has moved to another repo. Might be nice to add it some day if you can find that repo.

**Comment by cevich** [comment]
> the solution is to move the debian-tar timebomb up a few lines

Crap, I'm sorry you spent time on that, I thought I had force-pushed that exact fix. Apparently not, I'm really sorry :cry:

**Comment by cevich** [comment]
Confirmed, [moving the grub-common installation fixed the tar upgrade problem](https://cirrus-ci.com/task/6078445908656128?logs=main#L3377) (as in, we ***do not*** want tar to be upgraded).

**Comment by edsantiago** [comment]
Comparing to the most recent (broken tar) VM build, to minimize clutter:

- 20240506t132946z-f40f39d13
- 20240502t200454z-f40f39d13 &#x21D1; (built in #349)

|  |*debian*|*prior-fedora*|*fedora*|*fedora-aws*|*rawhide*|
|----|----|----|----|----|----|
|*base*         | 13.2            | 39-1.5          | Generic         | ?               | 41-0            |
|               | 13.1&nbsp;&#x21D1; |                |                |                |                |
|*kernel*       | 6.7.12-1        | 6.8.8-200       | 6.8.9-300       | 6.8.8-300       | 6.9.0-0.rc6.20240503gitf03359bca01b.56 |
|               |                | 6.8.7-200&nbsp;&#x21D1; | 6.8.8-300&nbsp;&#x21D1; | 6.8.7-300&nbsp;&#x21D1; | 6.9.0-0.rc6.52&nbsp;&#x21D1; |
|*netavark*     | 1.4.0-4.1       | 1.10.3-1        | 1.10.3-3        | 1.10.3-3        | 1.10.3-3        |
|               | 1.4.0-4&nbsp;&#x21D1; |                |                |                |                |
|*crun*         | 1.15-1          | 1.14.4-1        | 1.15-1          | 1.14.4-1        | 1.15-1          |
|               | 1.14.4-1&nbsp;&#x21D1; |                | 1.14.4-1&nbsp;&#x21D1; |                | 1.14.4-5&nbsp;&#x21D1; |
|*tar*          | 1.34+dfsg-1.2+deb12u1 | 1.35-2          | 1.35-3          | 1.35-3          | 1.35-3          |
|               | 1.35+dfsg-3&nbsp;&#x21D1; |                |                |                |                |

Tar is back to a good version. Let's try this.

**Comment by cevich** [comment]
@Luap99 @sbrivio-rh with this new `c20240506t132946z-f40f39d13` it seems we're still hitting the "permission-denied" problem in [the Debian podman integration tests (annotated log)](https://api.cirrus-ci.com/v1/artifact/task/6081697266008064/html/int-podman-debian-13-rootless-host-sqlite.log.html).  Since I [most everything else is passing](https://cirrus-ci.com/build/6601781365440512) (including Debian rootless system tests), I'm thinking it's time to toss an "it's broken" issue and disable that part of the matrix.  At least until it can be more deeply investigated.  Any objections?

**Comment by Luap99** [comment]
> @Luap99 @sbrivio-rh with this new `c20240506t132946z-f40f39d13` it seems we're still hitting the "permission-denied" problem in [the Debian podman integration tests (annotated log)](https://api.cirrus-ci.com/v1/artifact/task/6081697266008064/html/int-podman-debian-13-rootless-host-sqlite.log.html). Since I [most everything else is passing](https://cirrus-ci.com/build/6601781365440512) (including Debian rootless system tests), I'm thinking it's time to toss an "it's broken" issue and disable that part of the matrix. At least until it can be more deeply investigated. Any objections?

System tests pass because they default locations (so it is good to know that defaults works). I think we are very sure that /var/tmp is blocked by the profile but if the fix isn't working then I am fine to disable it for now. My tmpfs change will move it back to /tmp/... again so I can enable it back there without figuring apparmor out soon anyway

**Review by Luap99 — COMMENTED** [review]
LGTM

**Comment by Luap99** [comment]
Podman PR https://github.com/containers/podman/pull/22549 seems to be green, are we good merging this? I like to rebase my tmpfs work onto this .

**Comment by cevich** [comment]
@Luap99 Yes, pending a :+1: from @edsantiago I'm all set on this and https://github.com/containers/podman/pull/22549

Merging this will also help out Lokesh on his Windows-image update PR.

**Comment by edsantiago** [comment]
It's been an exhausting day. I don't have the brainpower to re-review the CentOS diffs, but AFAICT those have not been touched since my last review. And even if I missed something then, I don't think that portion is critical to the majority of the containers projects.

I do have the brainpower to re-review all the other commits, and have done so. I have quibbles about some some comments, but none of them are worth blocking Paul for. His tmpdir work is, IMO, really promising and my review tonight can give him a few extra hours to experiment. So, LGTM. Thank you @cevich, this was a really fierce one. Special thanks for breaking up your commits.

**Comment by cevich** [comment]
> AFAICT those have not been touched since my last review.

Indeed, no changes were made there.

Thanks for your patients and brainpower contributions @edsantiago

**Comment by sbrivio-rh** [comment]
> Thanks @sbrivio-rh.
> 
> > but I think it would be good to drop the workaround
> 
> Yep, I put a thing in place to remind us to circle back here in a month or so.

@cevich I (almost?) forgot to notify you: this is now fixed in Debian unstable (currently [0.0~git20240607.8a83b53-1](https://packages.debian.org/source/unstable/passt)), as well as Debian testing (same current version).

And by the way of `/tmp` and `/var/tmp`, @edsantiago and @luap99, if you missed it: [Debian's /tmpest in a teapot](
https://lwn.net/Articles/975565/). Long story short: with a fresh Debian unstable image, `/tmp` should now be a `tmpfs` just like it is on Fedora (I haven't tried, though).

**Comment by Luap99** [comment]
> And by the way of /tmp and /var/tmp, @edsantiago and @Luap99, if you missed it: [Debian's /tmpest in a teapot](https://lwn.net/Articles/975565/). Long story short: with a fresh Debian unstable image, /tmp should now be a tmpfs just like it is on Fedora (I haven't tried, though).

Yes I saw that. We have it configured now and it should stay that way as I increased the size limit to be higher than the default (unused memory is wasted memory after all).

**Pull request merged**